File: packages/test-equivalence-matchers/src/boundaries/nodes/convenience/StateVariable.ts
import { CONTEXT_VARIABLE_TYPE } from '@contract-case/case-entities-internal';
import { AnyMatcher } from '../base';

/**
 * Matches the content of a variable that comes from a state. See [state
 * definitions](https://case.contract-testing.io/docs/defining-contracts/state-definitions)
 * and [state
 * handlers](https://case.contract-testing.io/docs/reference/state-handlers) for
 * more details.
 */
export class StateVariable extends AnyMatcher {
  /** @internal */
  readonly '_case:matcher:type': typeof CONTEXT_VARIABLE_TYPE;

  /** @internal */
  readonly '_case:matcher:variableName': string;

  /**
   * @param name - The name of the variable
   */
  constructor(name: string) {
    super(CONTEXT_VARIABLE_TYPE);
    this['_case:matcher:variableName'] = name;
  }

  /**
   * For non-TypeScript implementations (see [AnyMatcher.toJSON()](#\@case-contract-testing/test-equivalence-matchers.AnyMatcher.toJSON))
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/convenience/StateVariable.java
package io.contract_testing.contractcase.dsl.matchers.convenience;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import java.lang.String;
import javax.annotation.Generated;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * Matches the content of a variable that comes from a state. See <a href="https://case.contract-testing.io/docs/defining-contracts/state-definitions">state definitions</a> and <a href="https://case.contract-testing.io/docs/reference/state-handlers">state handlers</a> for more details.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class StateVariable<M> implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * The name of the variable
   */
  @Getter
  @JsonProperty("_case:matcher:variableName")
  private final String variableName;

  @Builder
  public StateVariable(@NotNull final String variableName) {
    this.type = "_case:ContextVariable";
    this.variableName = variableName;
  }
}
